### PR TITLE
gherkin: Restore native JavaScript implementation (WIP)

### PR DESCRIPTION
**Pull request by badeball**
## Summary

Initial attempt at resurrecting the native JavaScript implementation of Gherkin.

## Details

1. It's essentially a checkout of the implementation as of #353.

2. It adds the implementation as a `legacy` \* property on the exported member, as well as a `--use-legacy-implementation` flag in the executable.

3. It updates the legacy implementation in order to satisfy all tests added since back then.

\* I've not given much thought into the name `legacy`, though it works for me personally.

## Motivation and Context

We've (@aslakhellesoy and myself) briefly talked about this on Slack. Right now, I'm first and foremost wondering if this is at all what you had in mind.

Secondly, if that is the case, I'm wondering what the next logical step is. Would this additon merely be temporary until other mature or is the shared executable approach being abandoned for JavaScript? I believe this affects EG. the value / effort ratio of converting everything to TypeScript.

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
I've started looking at this. Solid work @badeball ! I'm going to port the js to TypeScript and then merge and make a new release - ok?

**Comment by badeball** [comment]
Sounds like a plan! I might be able to help with that as well, if you already have stuff on your plate.

File: gherkin/javascript/src/legacy/gherkin/generate_events.js [hunk 1]
```diff
@@ -17,55 +35,50 @@ function generateEvents(data, uri, types, language) {
...
+          }
         }
       })
     }
 
     if (!types['gherkin-document'] && !types['pickle'])
       return result
 
-    var gherkinDocument = parser.parse(data, language)
+    var gherkinDocument = parser.parse(data, new TokenMatcher(language))
```
Comment (badeball) [inline gherkin/javascript/src/legacy/gherkin/generate_events.js]: Btw, notice this change. It turns out that some of the tests for the [EventStream](https://github.com/cucumber/cucumber/blob/gherkin-v5.1.0/gherkin/javascript/test/stream/event_stream_test.js) (v5.1.0) weren't actually run as part of CI. Hence this probably went unnoticed.

**Comment by ehuelsmann** [comment]
Hi, where is this slack channel? I've not found it in docs in the repo. I'd like to join the discussion, but have no way to know where to enter "the building".

**Comment by aslakhellesoy** [comment]
> Hi, where is this slack channel?

@ehuelsmann you'll find it at https://cucumber.io/support

**Comment by aslakhellesoy** [comment]
Hi @badeball,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**